### PR TITLE
Fix BSR->Dense Batched Bug

### PR DESCRIPTION
**Pull request by amjames**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82121
* #82122
* #80781
* __->__ #82120
* #82119

A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general.

### CODE REVIEW

**Comment by amjames** [comment]
@bhosmer @nikitaved @pearu This is ready for review

File: aten/src/ATen/native/TensorConversions.cpp [hunk 1]
```diff
@@ -400,10 +400,10 @@ Tensor sparse_compressed_to_dense(
     if (self.dim() > 3) {
       // Flatten batch dims
       auto n_batch_dim = self.dim() - 2;
-      crow_indices = crow_indices.flatten(0, n_batch_dim);
-      col_indices = col_indices.flatten(0, n_batch_dim);
-      values = values.flatten(0, n_batch_dim);
-      dense = dense.flatten(0, n_batch_dim);
+      crow_indices = crow_indices.flatten(0, n_batch_dim - 1);
```
Comment (nikitaved) [inline aten/src/ATen/native/TensorConversions.cpp:403]: alternatively, we can do `flatten(0, -2)` to avoid any explicit batch dims :)

**Review by nikitaved — APPROVED** [review]
Looks good!

**Review by bhosmer — APPROVED** [review]

**Pull request closed without merging**